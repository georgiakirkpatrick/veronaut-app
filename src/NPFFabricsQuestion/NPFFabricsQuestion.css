.NPFFabricsQuestion p {
    padding: 0;
}

.NPFFabricsQuestion .FormCheckboxSection {
    margin: 1vh 0;
}

.NPFFabricsQuestion .FormCheckboxSection p {
    padding: 1vh 0;
}

.NPFFabricsQuestion .FormCheckboxSection__option {
    padding: 1vh;
}

@media screen and (min-width: 768px) {
    .NPFFabricsQuestion .FormFieldset {
        padding: 2vh 5vw;
    }
}

@media screen and (min-width: 1024px) {
    .NPFFabricsQuestion .FormCheckboxSection p {
        padding: 0;
    }

    .NPFFabricsQuestion .FormFieldset {
        padding: 5vh 5vw;
    }
}

@media screen and (min-width: 1024px) and (min-height: 1366px) {
    .NPFFabricsQuestion .FormCheckboxSection p {
        padding: 0;
    }
}

@media screen and (min-width: 1440px) {
    .NPFFabricsQuestion .FormFieldset {
        padding: 5vh 3vw;
    }
}

@media screen and (min-width: 2560px) {
    .NPFFabricsQuestion .FormFieldset {
        padding: 3vh 3vw;
    }
}
