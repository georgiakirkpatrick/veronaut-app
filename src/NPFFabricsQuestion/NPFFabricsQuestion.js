import React from 'react'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormPage from '../FormPage/FormPage'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'

const NPFFabricsQuestion = props => {
    const {
        currentPage,
        linCheck,
        primCheck,
        setBackPageTurns,
        secCheck,
        setLinBackPageTurns,
        setLinCheck,
        setPage,
        setPrimCheck,
        setPrimPageTurns,
        setSecCheck,
        setSecPageTurns
    } = props

    const linCheckOption = [
        {
            text: 'Yes',
            id: 'lining-fabric',
            name: 'lining-fabric',
            checked: linCheck
        }
    ]

    const nextButton = () => {
        if (secCheck && linCheck) {
            setPrimPageTurns(1)
            setSecPageTurns(1)
            setBackPageTurns(1)
        } else if (secCheck && !linCheck) {
            setPrimPageTurns(1)
            setSecPageTurns(2)
            setBackPageTurns(2)
        } else if (!secCheck && linCheck) {
            setLinBackPageTurns(2)
            setPrimPageTurns(2)
            setSecPageTurns(1)
            setBackPageTurns(1)
        } else if (!secCheck && !linCheck) {
            setPrimPageTurns(3)
            setBackPageTurns(3)
        }
        
        setPage(currentPage + 1)
    }

    const primCheckOption = [
        {
            text: 'Yes',
            id: 'primary-fabric',
            name: 'primary-fabric',
            checked: primCheck
        }
    ]
    
    const secCheckOption = [
        {
            text: 'Yes',
            id: 'secondary-fabric',
            name: 'secondary-fabric',
            checked: secCheck
        }
    ]


    return (
        <div id='fabric-question'>
            <FormPage title='Fabrics'>
                <FormPromptWithSub 
                    prompt='The primary fabric (or material) is the main fabric used in the garment or accessory.'
                />

                <FormCheckboxSection 
                    prompt='The primary fabric (or material) is required.'
                    options={primCheckOption}
                    selectedOptions={{'primary-fabric': primCheck}}
                    handleChange={() => setPrimCheck(primCheck)}
                />

                <FormPromptWithSub 
                    prompt='Garments and accessories sometimes include secondary fabrics as trims or accents.'
                />

                <FormCheckboxSection 
                    prompt='Does this product include a secondary fabric?'
                    options={secCheckOption}
                    selectedOptions={{'secondary-fabric': secCheck}}
                    handleChange={() => setSecCheck(!secCheck)}
                />

                <FormPromptWithSub 
                    prompt='Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.'
                />

                <FormCheckboxSection 
                    prompt='Does this product include a lining fabric?'
                    options={linCheckOption}
                    selectedOptions={{'lining-fabric': linCheck}}
                    handleChange={() => setLinCheck(!linCheck)}
                />
            </FormPage>

            <NPFFooter 
                buttons='prevNext' 
                previousButton={() => setPage(currentPage - 1)} 
                nextButton={() => nextButton()} 
            />
        </div>
    )    
}

NPFFabricsQuestion.defaultProps = {
    currentPage: 0,
    linCheck: false,
    primCheck: false,
    secCheck: false,
    setLinCheck: () => {},
    setPage: () => {},
    setPrimCheck: () => {},
    setPrimPageTurns: () => {},
    setSecCheck: () => {},
    setSecPageTurns: () => {}
}

export default NPFFabricsQuestion