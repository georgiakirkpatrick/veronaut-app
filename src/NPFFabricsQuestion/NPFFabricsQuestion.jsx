import React from 'react'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormFieldset from '../FormFieldset/FormFieldset'
import FormPage from '../FormPage/FormPage'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'
import './NPFFabricsQuestion.css'

const NPFFabricsQuestion = ({
  currentPage = 0,
  linCheck = false,
  primCheck = false,
  secCheck = false,
  setLinCheck = () => {},
  setPage = () => {},
  setPrimCheck = () => {},
  setPrimPageTurns = () => {},
  setSecCheck = () => {},
  setSecPageTurns = () => {}
}) => {
  const linCheckOption = [
    {
      text: 'Yes, the product has a lining fabric or material',
      id: 'lining-fabric',
      name: 'lining-fabric',
      checked: linCheck
    }
  ]

  const nextButton = () => {
    if (secCheck && linCheck) {
      setPrimPageTurns(1)
      setSecPageTurns(1)
      setBackPageTurns(1)
    } else if (secCheck && !linCheck) {
      setPrimPageTurns(1)
      setSecPageTurns(2)
      setBackPageTurns(2)
    } else if (!secCheck && linCheck) {
      setLinBackPageTurns(2)
      setPrimPageTurns(2)
      setSecPageTurns(1)
      setBackPageTurns(1)
    } else if (!secCheck && !linCheck) {
      setPrimPageTurns(3)
      setBackPageTurns(3)
    }
    
    setPage(currentPage + 1)
  }

  const primCheckOption = [
    {
      text: 'The product has a primary fabric or material',
      id: 'primary-fabric',
      name: 'primary-fabric',
      checked: primCheck
    }
  ]
  
  const secCheckOption = [
    {
      text: 'Yes, the product has a secondary fabric or material',
      id: 'secondary-fabric',
      name: 'secondary-fabric',
      checked: secCheck
    }
  ]

  return (
    <div id='fabric-question' className='NPFFabricsQuestion'>
      <FormPage title='Fabrics'>
        <FormPromptWithSub 
          prompt="Select the product's material types below"
        />

        <FormFieldset>
          {/* <FormPromptWithSub 
              prompt='The primary fabric or material is the main fabric used in the garment or accessory.'
              promptSubtitle='The primary fabric is required.'
          /> */}

          <FormCheckboxSection 
            prompt='The primary fabric or material is the main fabric used in the garment or accessory.  The primary fabric is required.'
            options={primCheckOption}
            selectedOptions={{'primary-fabric': primCheck}}
            handleChange={() => setPrimCheck(primCheck)}
          />

          {/* <FormPromptWithSub 
              promptSubtitle=''
          /> */}

          <FormCheckboxSection 
            prompt='Garments and accessories sometimes include secondary fabrics as trims or accents.  Does this product include a secondary fabric?'
            options={secCheckOption}
            selectedOptions={{'secondary-fabric': secCheck}}
            handleChange={() => setSecCheck(!secCheck)}
          />

          {/* <FormPromptWithSub 
              promptSubtitle='Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.'
          /> */}

          <FormCheckboxSection 
            prompt='Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.  Does this product include a lining fabric?'
            options={linCheckOption}
            selectedOptions={{'lining-fabric': linCheck}}
            handleChange={() => setLinCheck(!linCheck)}
          />
        </FormFieldset>
      </FormPage>

      <NPFFooter 
        buttons='prevNext' 
        previousButton={() => setPage(currentPage - 1)} 
        nextButton={() => nextButton()} 
      />
    </div>
  )    
}

export default NPFFabricsQuestion