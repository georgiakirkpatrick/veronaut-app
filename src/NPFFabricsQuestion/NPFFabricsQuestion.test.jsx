import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import NPFFabricsQuestion from './NPFFabricsQuestion'

describe('NPFFabricsQuestion', () => {
  it ('renders without crashing', () => {
    render(<NPFFabricsQuestion />)
    screen.debug()
  })

  it ('render the UI as expected', () => {
    const tree = render(<NPFFabricsQuestion />)
    expect(tree).toMatchSnapshot()
  })
})