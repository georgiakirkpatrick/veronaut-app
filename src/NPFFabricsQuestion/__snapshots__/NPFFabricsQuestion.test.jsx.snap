// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NPFFabricsQuestion > render the UI as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="NPFFabricsQuestion"
        id="fabric-question"
      >
        <section
          class="FormPage"
        >
          <h1>
            Fabrics
          </h1>
          <form
            autocomplete="off"
            id="form-page"
          >
            <div
              class="FormPromptWithSub"
            >
              <p
                class="FormPromptWithSub__prompt"
              >
                Select the product's material types below
              </p>
              <p
                class="FormPromptWithSub__subtitle"
              />
            </div>
            <fieldset
              class="FormFieldset"
            >
              <div />
              <div
                class="FormCheckboxSection"
              >
                <p>
                  The primary fabric or material is the main fabric used in the garment or accessory.  The primary fabric is required.
                </p>
                <div
                  class="FormCheckboxSection__options-container"
                >
                  <div
                    class="FormCheckboxSection__option"
                  >
                    <input
                      id="primary-fabric"
                      name="primary-fabric"
                      type="checkbox"
                    />
                    <label
                      for="primary-fabric"
                    >
                      The product has a primary fabric or material
                    </label>
                  </div>
                </div>
              </div>
              <div
                class="FormCheckboxSection"
              >
                <p>
                  Garments and accessories sometimes include secondary fabrics as trims or accents.  Does this product include a secondary fabric?
                </p>
                <div
                  class="FormCheckboxSection__options-container"
                >
                  <div
                    class="FormCheckboxSection__option"
                  >
                    <input
                      id="secondary-fabric"
                      name="secondary-fabric"
                      type="checkbox"
                    />
                    <label
                      for="secondary-fabric"
                    >
                      Yes, the product has a secondary fabric or material
                    </label>
                  </div>
                </div>
              </div>
              <div
                class="FormCheckboxSection"
              >
                <p>
                  Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.  Does this product include a lining fabric?
                </p>
                <div
                  class="FormCheckboxSection__options-container"
                >
                  <div
                    class="FormCheckboxSection__option"
                  >
                    <input
                      id="lining-fabric"
                      name="lining-fabric"
                      type="checkbox"
                    />
                    <label
                      for="lining-fabric"
                    >
                      Yes, the product has a lining fabric or material
                    </label>
                  </div>
                </div>
              </div>
            </fieldset>
          </form>
          <div />
        </section>
        <footer
          class="NPFFooter"
        >
          <button
            class="NPFFooter__previous"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-left NPFFooter__left-arrow"
              data-icon="angle-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
                fill="currentColor"
              />
            </svg>
            <span>
              PREVIOUS
            </span>
          </button>
          <button
            class="NPFFooter__next"
            type="button"
          >
            <span>
              NEXT
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right NPFFooter__right-arrow"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="NPFFabricsQuestion"
      id="fabric-question"
    >
      <section
        class="FormPage"
      >
        <h1>
          Fabrics
        </h1>
        <form
          autocomplete="off"
          id="form-page"
        >
          <div
            class="FormPromptWithSub"
          >
            <p
              class="FormPromptWithSub__prompt"
            >
              Select the product's material types below
            </p>
            <p
              class="FormPromptWithSub__subtitle"
            />
          </div>
          <fieldset
            class="FormFieldset"
          >
            <div />
            <div
              class="FormCheckboxSection"
            >
              <p>
                The primary fabric or material is the main fabric used in the garment or accessory.  The primary fabric is required.
              </p>
              <div
                class="FormCheckboxSection__options-container"
              >
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="primary-fabric"
                    name="primary-fabric"
                    type="checkbox"
                  />
                  <label
                    for="primary-fabric"
                  >
                    The product has a primary fabric or material
                  </label>
                </div>
              </div>
            </div>
            <div
              class="FormCheckboxSection"
            >
              <p>
                Garments and accessories sometimes include secondary fabrics as trims or accents.  Does this product include a secondary fabric?
              </p>
              <div
                class="FormCheckboxSection__options-container"
              >
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="secondary-fabric"
                    name="secondary-fabric"
                    type="checkbox"
                  />
                  <label
                    for="secondary-fabric"
                  >
                    Yes, the product has a secondary fabric or material
                  </label>
                </div>
              </div>
            </div>
            <div
              class="FormCheckboxSection"
            >
              <p>
                Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.  Does this product include a lining fabric?
              </p>
              <div
                class="FormCheckboxSection__options-container"
              >
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="lining-fabric"
                    name="lining-fabric"
                    type="checkbox"
                  />
                  <label
                    for="lining-fabric"
                  >
                    Yes, the product has a lining fabric or material
                  </label>
                </div>
              </div>
            </div>
          </fieldset>
        </form>
        <div />
      </section>
      <footer
        class="NPFFooter"
      >
        <button
          class="NPFFooter__previous"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-left NPFFooter__left-arrow"
            data-icon="angle-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              fill="currentColor"
            />
          </svg>
          <span>
            PREVIOUS
          </span>
        </button>
        <button
          class="NPFFooter__next"
          type="button"
        >
          <span>
            NEXT
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-right NPFFooter__right-arrow"
            data-icon="angle-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
              fill="currentColor"
            />
          </svg>
        </button>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
